### PR TITLE
Null check WebSocket Client

### PR DESCRIPTION
**Pull request by withakay**
Fix for the null reference errors mentioned in #171

### CODE REVIEW

File: src/IO.Ably.Shared/Transport/MsWebSocketConnection.cs [hunk 1]
```diff
@@ -113,20 +113,20 @@ public void SendData(byte[] data)
...
-            if (ClientWebSocket.State == WebSocketState.Open)
+            if (ClientWebSocket?.State == WebSocketState.Open)
             {
                 var sendTask = ClientWebSocket.SendAsync(data, type, true, token);
                 sendTask.ConfigureAwait(false);
                 return sendTask;
             }
 
-            Logger.Warning($"Trying to send message of type {type} when the socket is already closed. Ack for this message will fail shortly.");
+            Logger.Warning($"Trying to send message of type {type} when the socket is already closed or null. Ack for this message will fail shortly.");
```
Comment (mattheworiordan) [inline src/IO.Ably.Shared/Transport/MsWebSocketConnection.cs]: Not sure about that error message.  What does it meant to a user if a socket is `null`?

Comment (withakay) [reply src/IO.Ably.Shared/Transport/MsWebSocketConnection.cs]: Having given this some more thought, I can't think of a legitimate reason for the ClientWebSocket to be null here, so I think the thing to do is set the Connection State to Failed and raise an exception. WDYT?

**Comment by withakay** [comment]
@mattheworiordan I have updated the code to set an Error state that propagates up to the ConnectionManager, causing a Disconnected state to be set. If auto connect is set then the connection reconnects.
I have made some small adjustments to allow for testing and added a test to cover this. Let me know what you think?

**Comment by mattheworiordan** [comment]
Thanks, LGTM, 👍

**Pull request merged**